packages/casadi/patches/0003-update-serializing_stream_cpp.patch

```diff
@@ -0,0 +1,31 @@
+From 31f4659ad64c0d05161a484d203baf4ffef5edd4 Mon Sep 17 00:00:00 2001
+From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
+Date: Mon, 15 Jul 2024 18:03:48 +0530
+Subject: [PATCH 3/4] Disable SIZE_MAX check for Emscripten, part II
+
+In WebAssembly (currently 32-bit), SIZE_MAX is equal to UINT_MAX, which meant
+these functions weren't being compiled, so there was no way to serialize or
+deserialize unsigned int values, leading to compile errors when these functions
+were called elsewhere. On many 32-bit systems, size_t and unsigned int are the
+same size, so SIZE_MAX == UINT_MAX is a common case.
+
+---
+ serializing_stream.cpp | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/casadi/core/serializing_stream.cpp b/casadi/core/serializing_stream.cpp
+index f3101f84..dfc6b2e4 100644
+--- a/casadi/core/serializing_stream.cpp
++++ b/casadi/core/serializing_stream.cpp
+@@ -155,7 +155,7 @@ namespace casadi {
+       for (int j=0;j<4;++j) pack(c[j]);
+     }
+ 
+-#if SIZE_MAX != UINT_MAX
```

Comment:
This should be combined into the same patch as 0002.

Reply:
That makes sense; I was thinking of doing it as well. I couldn't get `patchutils` to merge the files for me, so I did it manually in 914c44fb807da189105908e5ed433cdc71cd885d.